### ISSUE TITLE
Redesign error instances for BMC network settings

### ISSUE THREAD

**Issue by alissaylee** [openbmc/phosphor-webui#14]
![network_error_states_before_saving](https://user-images.githubusercontent.com/42816433/45120301-4f777280-b123-11e8-9ec2-7a8ccbc60b53.jpg)
**Error instances BEFORE saving settings:**
- All error fields can be validated after 10 sec user finishes typing. 
- Each error field is accommodated with an error message, error icon, and red border for text field.
- Error fields validated for character format: Hostname, Mac Address, Default Gateway, IPV4 address, Gateway, DNS Server.
- Netmask Prefix Length error field can be validated for numeric value (stays between 0 - 32).

![network_leaving_before_saving](https://user-images.githubusercontent.com/42816433/45120984-599a7080-b125-11e8-850b-6df45c45b037.jpg)
- Pop-up message appears when user tries to leave page without saving settings. Leaving without saving will undo changes made.

---------------------------------------------------------------------------------------------

![network_save_with_errors](https://user-images.githubusercontent.com/42816433/45120701-8437f980-b124-11e8-9e77-45aa4c1e30c5.jpg)
![network_errors_to_fix](https://user-images.githubusercontent.com/42816433/45120751-ac275d00-b124-11e8-80ef-12eb9c47cd4f.jpg)
**Error instances trying to save settings with persisting errors**
- Pop-up message appears noting user cannot save with persisting errors.
- Fields with "Invalid format" or empty fields will be highlighted as errors.
- Error messages go away after 10 sec once user inputs valid formats and/or all fields are entered.

---------------------------------------------------------------------------------------------
![network_error_in_saving](https://user-images.githubusercontent.com/42816433/45120773-bea19680-b124-11e8-8447-98132210fb09.jpg)
**Error instances AFTER saving settings**
- Pop-up message appears noting saving changes was unsuccessful due to errors.
- Fields with "Invalid IP address/DNS server or connection problem" will be highlighted as errors. Once inputting new valid addresses, user can save settings again.

**Comment by alissaylee** [comment]
Public InVision link to BMC Network Settings: https://ibm.invisionapp.com/share/YSNTBXAJGUC#/318318937_Network_Assign_new_static_IP

**Comment by susantjasinski** [comment]
@edtanous Yes, we are looking at the whole GUI and looking for validation and notification improvements

**Comment by derick-montague** [comment]
@gtmills This feels like it is pretty granular and would be better specified as pattern in the style guide for the entire GUI. The InVision links referenced are in an archived project @jandraa. Is this issue something that we still need to track here?

**Comment by derick-montague** [comment]
This work is accounted for in #62. We will revisit this story and possibly pull GUI work out following the completion of #62.
